src/Microsoft.Health.Dicom.Blob/Features/Storage/BlobFileStore.cs

```diff
@@ -167,12 +176,14 @@ public async Task DeleteFileIfExistsAsync(long version, string partitionName, Ca
     }
 
     /// <inheritdoc />
-    public async Task<Stream> GetFileAsync(long version, string partitionName, CancellationToken cancellationToken)
+    public async Task<Stream> GetFileAsync(long version, Partition partition, FileProperties fileProperties, CancellationToken cancellationToken)
     {
+        EnsureArg.IsNotNull(partition, nameof(partition));
 
-        BlockBlobClient blobClient = GetInstanceBlockBlobClient(version, partitionName);
+        BlockBlobClient blobClient = GetInstanceBlockBlobClient(version, partition, fileProperties);
 
         var blobOpenReadOptions = new BlobOpenReadOptions(allowModifications: false);
+        blobOpenReadOptions.Conditions = _blobClient.GetConditions(fileProperties);
```

Comment:
So for older DICOM services, which doesn't FileProperties populated, it will be null and also the Etag value will be Null right? I understand for update we have to block. Do we also want to block WADO and rendered API too?

The PR changes are not consistent, like some APIs are blocked but others are not.I know this PR is scoped to Update operation, but we are changing WADO and Rendered API here